src/app/Main.java

```diff
@@ -56,27 +60,52 @@ public static void main(String[] args) {
                     String userNameReservation = sn.next();
                     System.out.println("type lastname of user");
                     String userLastnameReservation = sn.next();
-                    boolean usuarioEncontrado = userService.userExists(userNameReservation, userLastnameReservation);
+                    boolean userFound = userService.userExists(userNameReservation, userLastnameReservation);
 
                     System.out.println("type name of book");
                     String nameBookReservation = sn.next();
                     System.out.println("type name of author");
                     String nameAuthorBookReservation = sn.next();
                     System.out.println("type year of book");
                     int yearBookReservation = sn.nextInt();
-                    boolean libroEncontrado = bookService.bookExists(nameBookReservation, nameAuthorBookReservation, yearBookReservation);
+                    boolean bookFound = bookService.bookExists(nameBookReservation, nameAuthorBookReservation, yearBookReservation);
 
-                    if (usuarioEncontrado && !libroEncontrado) {
+                    if (userFound && bookFound) {
                         reservationService.createReservation(userNameReservation, userLastnameReservation, nameBookReservation, nameAuthorBookReservation, yearBookReservation);
+                        System.out.println("reservation has been created");
+                    } else {
+                        if (!userFound) {
+                            System.out.println("User not found");
+                            System.out.println("do you want to create it?");
+                            System.out.println("1 YES");
+                            System.out.println("2 NO");
+                            option = sn.nextInt();
+                            switch (option) {
+                                case 1:
+                                    userService.createUser(userNameReservation, userLastnameReservation);
+                                case 2:
+                                    continue;
+                            }
+                        }
                     }
+
                     break;
 
                 case 4:
                     System.out.println("type name of user");
                     String userNameFound = sn.next();
                     System.out.println("type lastname of user");
                     String userLastnameFound = sn.next();
-                    reservationService.searchReservations(userNameFound, userLastnameFound);
+                    boolean validationUserReservation = userService.userExists(userNameFound, userLastnameFound);
```

Comment:
This logic should live in the Reservation service.

This is part of your business logic, means you cant look for a reservation for a user that does not exists.
Move this check to a method within the reservation service